### PR TITLE
generic: SYCL: convolution/deconvolution/softmax: add missing checks

### PR DESCRIPTION
**Pull request by t4c1**
Adds some previously missing checks on the SYCL implementations of convolution, deconvolution and softmax.

### CODE REVIEW

**Review by mgouicem — APPROVED** [review]

**Review by avmanerikar — APPROVED** [review]

File: src/gpu/generic/sycl/ref_softmax.hpp [hunk 1]
```diff
@@ -48,6 +48,7 @@ struct ref_sycl_softmax_fwd_t : public gpu::generic::sycl::primitive_t {
                     && sycl_post_ops_t::post_ops_ok(attr(), true, false)
                     && set_default_formats() == status::success
                     && attr_.set_default_formats(dst_md()) == status::success
+                    && check_formats(src_md(), dst_md())
```
Comment (sgeor255) [inline src/gpu/generic/sycl/ref_softmax.hpp:51]: I believe in the backward pass we should check `diff_src`, `diff_dst` & `dst`, `src` is not used in the backward pass.

File: src/gpu/generic/sycl/ref_deconvolution.hpp [hunk 2]
```diff
@@ -59,7 +59,8 @@ struct ref_deconvolution_bwd_weights_t
                             data_d, diff_weights_d, diff_dst_d)
                     && attr()->has_default_values(sm::scales_runtime
                             | sm::zero_points_runtime | sm::post_ops
-                            | sm::sum_dt);
+                            | sm::sum_dt)
```
Comment (mgouicem) [inline src/gpu/generic/sycl/ref_deconvolution.hpp:62]: (not related to this PR) there is no scaling for bwd pass.

File: src/gpu/generic/sycl/ref_convolution.hpp [hunk 3]
```diff
@@ -203,7 +205,8 @@ struct ref_convolution_bwd_weights_t : public gpu::generic::sycl::primitive_t {
                     && attr()->has_default_values(sm::scales_runtime
                             | sm::zero_points_runtime | sm::sum_dt)
                     && IMPLICATION(!attr()->scales_.has_default_values(),
-                            attr_scales_ok());
+                            attr_scales_ok())
```
Comment (mgouicem) [inline src/gpu/generic/sycl/ref_convolution.hpp:208]: (not related to this PR) there should be no scaling for bwd pass.

File: src/gpu/generic/sycl/ref_softmax.hpp (hunk 1 above)
Comment (t4c1) [reply src/gpu/generic/sycl/ref_softmax.hpp]: fixed

File: src/gpu/generic/sycl/ref_deconvolution.hpp (hunk 2 above)
Comment (t4c1) [reply src/gpu/generic/sycl/ref_deconvolution.hpp]: fixed, along with post ops

File: src/gpu/generic/sycl/ref_convolution.hpp (hunk 3 above)
Comment (t4c1) [reply src/gpu/generic/sycl/ref_convolution.hpp]: fixed, along with post ops

**Comment by mgouicem** [comment]
Could you please resolve the conflict? (I think it is just a matter of adding both checks on scales_dt and algo, but would prefer if you could check :))

**Pull request merged**